Dockerfile

```diff
@@ -0,0 +1,63 @@
+# For more information, please refer to https://aka.ms/vscode-docker-python
+FROM python:3.8-slim-buster as python-base
+
+# Keeps Python from generating .pyc files in the container
+ENV PYTHONDONTWRITEBYTECODE=1 \
+# Turns off buffering for easier container logging
+PYTHONUNBUFFERED=1 \
+POETRY_VERSION=1.0.0 \
+ # make poetry install to this location
+POETRY_HOME=".poetry/bin"
+# paths
+# this is where our requirements + virtual environment will live
+#PYSETUP_PATH="/opt/pysetup"
+
+# `builder-base` stage is used to build deps 
+FROM python-base as builder-base
+RUN apt-get update \
+    && apt-get install --no-install-recommends -y \
+        # deps for installing poetry
+        curl \
+        # deps for building python deps
+        build-essential
+
+#RUN pip install "poetry==$POETRY_VERSION"
+
+RUN curl -sSL https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py | python -
+WORKDIR /todo_app
+#WORKDIR $PYSETUP_PATH
+COPY poetry.lock pyproject.toml /todo_app/
+
+ENV PATH="$POETRY_HOME:$PATH"
+
+RUN poetry config virtualenvs.create false \
+  && poetry install --no-dev
+
+
+
+FROM base as production
+
+COPY . /todo_app
+# copy in our built poetry + venv
+COPY --from=base $POETRY_HOME $POETRY_HOME
+#COPY --from=base $PYSETUP_PATH $PYSETUP_PATH
```

Comment:
If you doing `FROM base ...` then you are building on top of that image, so you already have all of the same files. No need to COPY anything in explicitly.